### ISSUE TITLE
Updating multiple records matching different criteria

### ISSUE THREAD

**Issue by skgbanga** [msiemens/tinydb#346]
Hello,

Currently tinydb supports inserting multiple entries in the database like this:

```py
            records = []                                                                                                                    
            for name, value in new.items():                                                                                                                                                                                       
                record = {                                                                                                                  
                    "name": name,                                                                                                                                                                                        
                    "value": value,                                                                                               
                }                                                                                                                           
                records.append(record)                                                                                                                                                          
            if records:                                                                                                                     
                self._db.insert_multiple(records)
```


But when we have to update multiple entries, we have to do it one by one like this:                                                                                        
```py                                                                                                                                                                                                                                           
            q = Query()                                                                                                                 
            for name, value in update.items():                                                                                   
                self._db.update({"value": value}, q.name == name)  
```

In my application, updating entries in the database one by one is taking a massive amount of time. Is there a way in which I can do a batch (multiple) updates in one go?


Thanks!

**Comment by msiemens** [comment]
That's an good idea. Right now there are no batch updates in TinyDB but I can imagine implementing something like a `update_multiple` just as `insert` has `insert_multiple`. The only thing I'm not sure about whether or not to also allow updating multiple documents by doc_id. I will have to think about this a little more.

I've pushed an experimental implementation to the [`feat/update_multiple`](https://github.com/msiemens/tinydb/compare/feat/update_multiple) branch

**Comment by msiemens** [comment]
This is now published in v4.3.0 🙂

**Comment by msiemens** [comment]
Closing due to inactivity.
